sdk/keyvault/keyvault-keys/README.md

```diff
@@ -239,18 +240,38 @@ _deleted_ key. A deleted key can't be updated. They can only be either
 read, recovered or purged.
 
 ```javascript
-await client.deleteKey(keyName);
+const poller = await client.beginDeleteKey(keyName)
+await poller.pollUntilDone();
 
 // If soft-delete is enabled, we can eventually do:
 await client.getDeletedKey(keyName);
+
 // Deleted keys can also be recovered or purged:
-await client.recoverDeletedKey(keyName);
+// recoverDeletedKey also returns a poller, just like beginDeleteKey.
+const recoverPoller = await client.beginRecoverDeletedKey(keyName)
+const recoverPoller.pollUntilDone();
+
 // await client.purgeDeletedKey(keyName);
 ```
 
-Since the deletion of a key won't happen instantly, some time is needed
-after the `deleteKey` method is called before the deleted key is
-available to be read, recovered or purged.
+Since Keys take some time to get fully deleted, `beginDeleteKey`
+returns a Poller object that keeps track of the underlying Long Running
+Operation according to our guidelines:
+https://azure.github.io/azure-sdk/typescript_design.html#ts-lro
+
+Once you receive the poller, you'll be able to either run individual service
+calls until the key is deleted, or wait until the process is done:
+
+```typescript
+const poller = await client.beginDeleteKey(certificateName, certificatePolicy);
+
+await poller.poll(); // One service call
```

Comment:
In this particular case does .poll() map to a specific service call so they could get a better idea of what the "one service call" is?

As a user reading this I feel like it would give me a better idea of what you're doing from a logical perspective.

Reply:
I'll add a better comment!